birdhouse/components/weaver/docker-compose-extra.yml

```diff
@@ -0,0 +1,178 @@
+    container_name: ${WEAVER_MANAGER_NAME}
+    # NOTE: Override the usual Weaver-Manager image by one that is built inline with extra dependency to 'dockerize'
+    #   The 'dockerize' CLI utility allows to wait until each WPS GetCapabilities responds correctly, such that once
+    #   Weaver is started, they will all be able to request them and register the providers correctly.
+    #   This is needed because simply adding 'depends_on' only ensures that containers are 'running', but startup of
+    #   WPS applications themselves are not necessarily completed. Successful HTTP responses ensure they are 'ready'.
+    #image: pavics/weaver:${WEAVER_VERSION}-manager
+    image: pavics/weaver:${WEAVER_VERSION}-wait-wps
+    build:
+      context: ./components/weaver
+      dockerfile: Dockerfile
+      args:
+        WEAVER_VERSION: ${WEAVER_VERSION}
+    ports:
+      - "4001:4001"
+    environment:
+      HOSTNAME: ${PAVICS_FQDN}
+      FORWARDED_ALLOW_IPS: "*"
+    #env_file:
+    #  - ./config/mongodb/credentials.env
+    # ensure other referenced services are created beforehand to avoid booting timing issues
+    depends_on:
+      - catalog
+      - finch
+      - flyingpigeon
+      - hummingbird
+      - malleefowl
+      - raven
+      #- wait-for-wps
+      - mongodb
+    links:
+      - mongodb
+      - catalog
+      - finch
+      - flyingpigeon
+      - hummingbird
+      - malleefowl
+      - raven
+    networks:
+      - default
+      - mongodb
+    volumes:
+      - ./components/weaver/config/weaver/weaver.ini:/opt/local/src/weaver/config/weaver.ini:ro
+      - ./components/weaver/config/weaver/data_sources.json:/opt/local/src/weaver/config/data_sources.json:ro
+      - ./components/weaver/config/weaver/request_options.yml:/opt/local/src/weaver/config/request_options.yml:ro
+      - ./components/weaver/config/weaver/wps_processes.yml:/opt/local/src/weaver/config/wps_processes.yml:ro
+      # WARNING: see detail in 'worker' definition about below mount location
+      - ${WEAVER_WPS_OUTPUTS_DIR}:${WEAVER_WPS_OUTPUTS_DIR}
+    restart: always
+    logging: *default-logging
+    # NOTE: see above 'image' and 'build' sections, start Weaver only after WPS providers can respond
+    command: ["bash", "-c",
+        "/usr/local/bin/dockerize
+          -wait 'http://catalog:80/pywps?service=WPS&request=GetCapabilities'
+          -wait 'http://finch:5000/wps?service=WPS&request=GetCapabilities'
+          -wait 'http://flyingpigeon:8093/wps?service=WPS&request=GetCapabilities'
+          -wait 'http://hummingbird:8097/wps?service=WPS&request=GetCapabilities'
+          -wait 'http://malleefowl:8091/wps?service=WPS&request=GetCapabilities'
+          -wait 'http://raven:9099/wps?service=WPS&request=GetCapabilities'
+          -timeout 30s
+        && pserve '/opt/local/src/weaver/config/weaver.ini'"
+    ]
+
+  # Image 'worker' is the job runner of Weaver, possibly spawning Docker applications according to process definition
+  worker:
```

Comment:
Could this be named `weaver-worker` and not just plain `worker` to be clearer?  We might eventually have a `dask-worker` container in the stack.